platform/windows/detect.py

```diff
@@ -122,15 +135,15 @@ def setup_msvc_manual(env):
 
     # If building for 64bit architecture, disable assembly optimisations for 32 bit builds (theora as of writing)... vc compiler for 64bit can not compile _asm
     if compiler_version_str == "amd64" or compiler_version_str == "x86_amd64":
-        env["bits"] = "64"
+        env["arch"] = "x86_64"
         env["x86_libtheora_opt_vc"] = False
-        print("Compiled program architecture will be a 64 bit executable (forcing bits=64).")
+        print("Compiled program architecture will be a 64 bit executable (forcing arch=x86_64).")
     elif compiler_version_str == "x86" or compiler_version_str == "amd64_x86":
-        print("Compiled program architecture will be a 32 bit executable. (forcing bits=32).")
+        print("Compiled program architecture will be a 32 bit executable (forcing arch=x86_32).")
     else:
         print(
-            "Failed to manually detect MSVC compiler architecture version... Defaulting to 32bit executable settings"
-            " (forcing bits=32). Compilation attempt will continue, but SCons can not detect for what architecture this"
+            "Failed to manually detect MSVC compiler architecture version... Defaulting to x86 32-bit executable settings"
+            " (forcing arch=x86_32). Compilation attempt will continue, but SCons can not detect for what architecture this"
             " build is compiled for. You should check your settings/compilation setup, or avoid setting VCINSTALLDIR."
         )
```

Comment:
This should just fail, there's no point to continue if compiler is not present.

Reply:
The print is no longer accurate then, since it says explicitly that it's falling back to x86 32-bit settings.